### PR TITLE
Combine DataLoadStatus and DataBackupStatus into OperationStatus

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
Currently, dataload and databackup have crd different `status` type, combine struct `DataLoadStatus` and `DataBackupStatus` into `OperationStatus`

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅳ. Describe how to verify it
manual test

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: api/v1alpha1/status.go [hunk 1]
```diff
@@ -130,6 +130,19 @@ type RuntimeStatus struct {
...
+type OperationStatus struct {
+	// Phase describes current phase of DataBackup
+	Phase common.Phase `json:"phase"`
+	// Duration tell user how much time was spent to operation
+	Duration string `json:"duration"`
+	// Conditions consists of transition information on DataBackup's Phase
+	Conditions []Condition `json:"conditions"`
+
+	// Props data operation customized name-value
+	Props map[string]string `json:"props"`
```
Comment (cheyang) [inline api/v1alpha1/status.go:143]: I suggest renaming `props`  to `infos`. Thx.

Comment (xliuqq) [reply api/v1alpha1/status.go]: fixed,

**Comment by cheyang** [comment]
@xliuqq please fix the build, thanks.

**Comment by xliuqq** [comment]
> @xliuqq please fix the build, thanks.

The build error is not related to this PR.

File: api/v1alpha1/status.go [hunk 2]
```diff
@@ -130,6 +130,19 @@ type RuntimeStatus struct {
 	Mounts []Mount `json:"mounts,omitempty"`
 }
 
+// OperationStatus defines the observed state of certain Data Operation
+type OperationStatus struct {
+	// Phase describes current phase of DataBackup
```
Comment (zwwhdls) [inline api/v1alpha1/status.go:135]: comments are not correct and results in the description of CRD not correct.

File: config/crd/bases/data.fluid.io_dataloads.yaml [hunk 3]
```diff
@@ -106,10 +106,11 @@ spec:
                 type: array
             type: object
           status:
-            description: DataLoadStatus defines the observed state of DataLoad
+            description: OperationStatus defines the observed state of certain Data
+              Operation
             properties:
               conditions:
-                description: Conditions consists of transition information on DataLoad's
+                description: Conditions consists of transition information on DataBackup's
```
Comment (zwwhdls) [inline config/crd/bases/data.fluid.io_dataloads.yaml:113]: not `DataBackup's `

File: config/crd/bases/data.fluid.io_dataloads.yaml [hunk 4]
```diff
@@ -144,15 +145,20 @@ spec:
...
+                description: Duration tell user how much time was spent to operation
                 type: string
+              infos:
+                additionalProperties:
+                  type: string
+                description: Infos data operation customized name-value
+                type: object
               phase:
-                description: Phase describes current phase of DataLoad
+                description: Phase describes current phase of DataBackup
```
Comment (zwwhdls) [inline config/crd/bases/data.fluid.io_dataloads.yaml:156]: ditto

File: api/v1alpha1/status.go [hunk 5]
```diff
@@ -130,6 +130,19 @@ type RuntimeStatus struct {
...
+	Phase common.Phase `json:"phase"`
+	// Duration tell user how much time was spent to operation
+	Duration string `json:"duration"`
+	// Conditions consists of transition information on DataBackup's Phase
```
Comment (zwwhdls) [inline api/v1alpha1/status.go:139]: ditto

**Review by zwwhdls — CHANGES_REQUESTED** [review]

**Comment by TrafalgarZZZ** [comment]
> > @xliuqq please fix the build, thanks.
> 
> The build error is not related to this PR.
> 
> The sequence obtained by multiple traversals of map in Golang is different, but the thin fuse tests depend on this order which may cause in **occasional** errors.
> 
> Source code is 1）https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/transform_fuse_test.go#L453 2）https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/transform_fuse_test.go#L517 3）https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/transform_fuse.go#L194
> 
> @cheyang how to fix this thin fuse transform test?

@xliuqq That's weird. This should be fixed in [PR#2548](https://github.com/fluid-cloudnative/fluid/pull/2548/files#diff-2bbf625a7b62e7882daa53c2320bb865ab973bbbe8fcc5dcd53c544edfae6dabR497) by using `DeepEqualIgnoringSliceOrder`

**Comment by xliuqq** [comment]
> @xliuqq That's weird. This should be fixed in [PR#2548](https://github.com/fluid-cloudnative/fluid/pull/2548/files#diff-2bbf625a7b62e7882daa53c2320bb865ab973bbbe8fcc5dcd53c544edfae6dabR497) by using `DeepEqualIgnoringSliceOrder`

See https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/transform_fuse_test.go#L453
The source and target **list** are ["a=b","c=d"] and ["c=d","a=b"], they are **equal**.
But the source and target **string** are "a=b,c=d" and "c=d,a=b", they are not **equal**.

File: api/v1alpha1/status.go (hunk 2 above)
Comment (xliuqq) [reply api/v1alpha1/status.go]: Fixed. Thx.

File: api/v1alpha1/status.go (hunk 5 above)
Comment (xliuqq) [reply api/v1alpha1/status.go]: Fixed. Thx.

File: config/crd/bases/data.fluid.io_dataloads.yaml (hunk 4 above)
Comment (xliuqq) [reply config/crd/bases/data.fluid.io_dataloads.yaml]: Fixed. Thx.

File: config/crd/bases/data.fluid.io_dataloads.yaml (hunk 3 above)
Comment (xliuqq) [reply config/crd/bases/data.fluid.io_dataloads.yaml]: Fixed. Thx.

**Comment by cheyang** [comment]
> > @xliuqq That's weird. This should be fixed in [PR#2548](https://github.com/fluid-cloudnative/fluid/pull/2548/files#diff-2bbf625a7b62e7882daa53c2320bb865ab973bbbe8fcc5dcd53c544edfae6dabR497) by using `DeepEqualIgnoringSliceOrder`
> 
> @TrafalgarZZZ See https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/transform_fuse_test.go#L453 The source and target **list** are ["a=b","c=d"] and ["c=d","a=b"], they are **equal**. But the source and target **string** are "a=b,c=d" and "c=d,a=b", they are not **equal**.

the fix is https://github.com/fluid-cloudnative/fluid/pull/2607. Please review @xliuqq

**Review by cheyang — APPROVED** [review]

**Pull request merged**